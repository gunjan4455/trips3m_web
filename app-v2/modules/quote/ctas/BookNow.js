import React, { Component } from 'react';
import PropTypes from 'prop-types';

import EVENTS_ENUMS from 'modules/quote/EventTypes';

class BookNow extends Component {

  handleBookNow = () => {
    const { tripId, quoteId } = this.props;
    this.props.callbackHandler(EVENTS_ENUMS.BOOK_NOW, { tripId, quoteId });
  };

  render() {
    const { isQuoteCardCta } = this.props;
    return (
      <div className="flexFull">
        <button className={isQuoteCardCta ? 'block p15 text-left wfull at_bookNow' : 'wfull btn-pri-large ripple pl5 pr5 at_bookNow'} onClick={this.handleBookNow}>
          Book Now
        </button>
      </div>
    );
  }
}

BookNow.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  opsDetail: PropTypes.object,
  callbackHandler: PropTypes.func,
  trackSegment: PropTypes.func,
  isQuoteCardCta: PropTypes.bool
};

BookNow.defaultProps = {
  opsDetail: {},
  callbackHandler: () => {
  },
  trackSegment: () => {
  },
  isQuoteCardCta: false
};

export default BookNow;
