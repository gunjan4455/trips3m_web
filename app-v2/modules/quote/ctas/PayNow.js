import React from 'react';
import PropTypes from 'prop-types';

import EVENTS_ENUMS from "../EventTypes";

const CTA_TEXT = 'Pay Now';

const PayNow = ({ payNowUrl, ctaText, disablePayNow, callbackHandler, trackSegment }) => {
  const handlePayNow = () => {
    // trackSegment('Book Now clicked', 'Quote Price', 'Book Now');

    callbackHandler(EVENTS_ENUMS.PAY_NOW);
  };

  return (<div className="flexFull flex">
    {
      disablePayNow ?
        <div className="flexFull flex">
          <button className="wfull btn-pri-large ripple pl5 pr5" onClick={handlePayNow}>{ctaText}</button>
        </div> :
        <div className="flexFull flex"><a href={payNowUrl} className="wfull btn-filled-pri-large radius2 ripple">
          {ctaText}
        </a></div>
    }
  </div>);
};

PayNow.propTypes = {
  payNowUrl: PropTypes.string,
  ctaText: PropTypes.string,
  disablePayNow: PropTypes.bool,
  callbackHandler: PropTypes.func.isRequired,
  trackSegment: PropTypes.func
};

PayNow.defaultProps = {
  payNowUrl: '',
  ctaText: CTA_TEXT,
  disablePayNow: false,
  trackSegment: () => {
  }
};

export default PayNow;

