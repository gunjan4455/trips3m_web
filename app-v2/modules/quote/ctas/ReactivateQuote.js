import React from 'react';
import PropTypes from 'prop-types';

import EVENTS_ENUMS from 'modules/quote/EventTypes';

const CTA_TEXT = 'Reactivate Quote';

const ReactivateQuote = ({
                           tripId,
                           quoteId,
                           loading,
                           callbackHandler,
                           isQuoteCardCta
                         }) => (
  <div className="flexFull flex">
    {
      <button
        className={isQuoteCardCta ? 'block p15 text-left wfull' : 'wfull btn-pri-large ripple pl5 pr5'}
        disabled={loading}
        onClick={() => {
          callbackHandler(EVENTS_ENUMS.REACTIVATE_QUOTE, { tripId, quoteId });
        }}>
        {CTA_TEXT}
      </button>
    }
  </div>
);

ReactivateQuote.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  loading: PropTypes.bool,
  callbackHandler: PropTypes.func.isRequired,
  isQuoteCardCta: PropTypes.bool
};

ReactivateQuote.defaultProps = {
  isQuoteCardCta: false,
  loading: false
};

export default ReactivateQuote;

