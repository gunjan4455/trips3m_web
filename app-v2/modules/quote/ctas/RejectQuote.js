import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Modal from 'modules/shared/GenericModal';
import RejectQuotePopup from '../popups/rejectQuote/index';
import { rescue } from 'helpers/utils';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';

class RejectQuote extends Component {
  constructor(props) {
    super(props);

    this.state = {
      closeModal: false
    };
  }

  closePopup = () => {
    this.setState({ closeModal: true });
  };

  trackSegmentEvent = () => {
    rescue(() => {
      this.props.trackSegment(ORGANISM_CLICKED, 'Quotes section', 'Reject quote', '', '', this.props.quoteId, 'quote id');
    });
  };

  render() {
    const { quoteId, tripId, isQuoteCardCta, quoteRejectionReasons, callbackHandler, trackSegment } = this.props;
    return (
      <div className="flexFull flex">
        <Modal
          trigger={
            <button className={isQuoteCardCta ? 'block p15 fw7 text-left wfull at_rejectQuote' : 'wfull p15 fw7 cursorP at_rejectQuote'}
              onClick={this.trackSegmentEvent}>Reject Quote
            </button>
          }
          fullView={true}
          hasCustomClose={true}
          customCloseTrigger={this.state.closeModal}
          isTriggerFullView={true}
        >
          <RejectQuotePopup rejectQuoteReasons={quoteRejectionReasons}
                            closeModal={this.closePopup}
                            rejectQuoteHandler={callbackHandler}
                            tripId={tripId}
                            quoteId={quoteId}
                            loading={this.props.loading} />
        </Modal>
      </div>
    );
  }
}

RejectQuote.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  callbackHandler: PropTypes.func.isRequired,
  quoteRejectionReasons: PropTypes.object.isRequired,
  isQuoteCardCta: PropTypes.bool,
  loading: PropTypes.bool,
  trackSegment: PropTypes.func
};

RejectQuote.defaultProps = {
  isQuoteCardCta: false,
  loading: false
};

export default RejectQuote;
