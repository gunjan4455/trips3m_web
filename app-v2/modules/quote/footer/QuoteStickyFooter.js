import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import CancelBooking from 'modules/trip/ctas/CancelBooking';
import SwitchQuote from '../ctas/SwitchQuote';
import ExplainTheQuote from '../ctas/ExplainTheQuote';
import ViewInvoiceAndPay from '../ctas/ViewInvoiceAndPay';
import RejectQuote from '../ctas/RejectQuote';
import QUOTE_EVENTS_ENUMS from 'modules/quote/EventTypes';
import TRIP_EVENTS_ENUMS from 'modules/trip/ChildEventsType';
import Modal from 'modules/shared/GenericModal';
import BookNowPopup from '../popups/bookNow/index';
import { INFO_ROSTER } from 'constants/rosterTypes';
import { downloadQuoteUrl } from 'helpers/urlHelpers';
import { formatURL } from 'utils/parsers';
import { PDM_CATEGORY_EXPLAINQUOTE } from 'helpers/pdmHelpers';

const GFooterDiv = styled.div({
  minHeight: '48px',
  position: 'fixed',
  bottom: '0',
  left: '0',
  right: '0',
  justifyContent: 'space-between',
  zIndex: '9'
});

class QuoteStickyFooter extends Component {

  constructor(props) {
    super(props);

    this.state = {
      openModal: false,
      closeModal: true
    };

    this.callbackHandler = this.callbackHandler.bind(this);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (!this.props.rejectQuoteSuccess && nextProps.rejectQuoteSuccess) {
      this.props.showRoster({ type: INFO_ROSTER, message: 'Quote Has Been Rejected' });
    }

    else if (!this.props.reactivateQuoteSuccess && nextProps.reactivateQuoteSuccess) {
      this.props.showRoster({ type: INFO_ROSTER, message: 'Quote Reactivated Successfully!' });
    }
  }

  trackSegmentEvent = (event, cta, object = '') => {
    this.props.trackSegment({
      event,
      section: 'Bottom Sticky ',
      object: object,
      cta
    });
  };

  callbackHandler = (type, data = {}) => {
    data.tripId = this.props.tripId;
    data.quoteId = this.props.quoteId;

    switch (type) {
      case TRIP_EVENTS_ENUMS.CANCEL_TRIP:
        this.props.cancelBooking(data);
        break;
      case QUOTE_EVENTS_ENUMS.PAY_NOW:
        this.openBookNowPopup();
        this.trackSegmentEvent(ORGANISM_CLICKED, 'View Invoice & Pay');
        break;
      case QUOTE_EVENTS_ENUMS.REJECT_QUOTE:
        this.props.rejectQuote(data);
        this.trackSegmentEvent(ORGANISM_CLICKED, 'Reject Quote');
        break;
    }
  };

  openBookNowPopup = () => {
    this.setState({ openModal: true, closeModal: false });
  };

  closePopup = () => {
    this.setState({ closeModal: true, openModal: false });
  };

  handleChatClick = () => {
    const { setPdmCategory, showHidePdmView } = this.props;
    setPdmCategory(PDM_CATEGORY_EXPLAINQUOTE);
    showHidePdmView(true);
    this.trackSegmentEvent(ORGANISM_CLICKED, 'Explain the Quote');
  };

  render() {
    const ctas = [];
    const { tripId, quoteId, quoteStatus, currentQuote, convertedQuote, userDetails } = this.props;
    const shareLink = formatURL({ path: downloadQuoteUrl(quoteId) });
    if (quoteStatus.isRejected) {
      ctas.push(<ExplainTheQuote handleChatClick={this.handleChatClick} key="explain-the-quote" />);
    } else if (quoteStatus.isRequestCancelled) {
      ctas.push(<ExplainTheQuote handleChatClick={this.handleChatClick} key="explain-the-quote" />);
    } else if (quoteStatus.isCancelled) {
      if (convertedQuote.id) {
        ctas.push(<SwitchQuote tripId={tripId} quoteId={convertedQuote.id} key="switch_quote" />);
      }
    } else if (quoteStatus.isConverted) {
      if (currentQuote.installmentInfo.isAllPaymentReceived) {
        if (currentQuote.invoiceShowUrl) {
          ctas.push(<ViewInvoiceAndPay invoiceUrl={currentQuote.invoiceShowUrl} ctaText='View Invoice'
                                       key="cta-view-invoice" />);
        }
      } else {
        ctas.push(<CancelBooking callbackHandler={this.callbackHandler} tripId={tripId} key="cta-cancel-booking" />);
        ctas.push(<ViewInvoiceAndPay invoiceUrl={currentQuote.invoiceShowUrl} key="cta-view-invoice" />);
      }
    } else if (quoteStatus.isInvoiced) {
      ctas.push(<RejectQuote quoteRejectionReasons={this.props.quoteRejectionReasons}
                             callbackHandler={this.callbackHandler} tripId={tripId} quoteId={quoteId}
                             key="cta-reject-quote" />);
      ctas.push(<ViewInvoiceAndPay invoiceUrl={currentQuote.invoiceShowUrl} key="cta-view-invoice" />);
    }
    else {
      ctas.push(<ExplainTheQuote handleChatClick={this.handleChatClick} key="explain-the-quote" />);
    }

    if (!ctas.length) {
      return null;
    }

    return (
      <div>
        <GFooterDiv className="flex sbcw p8 bs6 at_quoteFooterSection">
          {
            ctas
          }
          <Modal
            trigger={null}
            fullView={true}
            isOpen={this.state.openModal}
            hasCustomClose={true}
            customCloseTrigger={this.state.closeModal}
            onRequestCloseCustom={this.closePopup}
          >
            <BookNowPopup
              opsDetail={this.props.currentQuote.opDetails}
              closeModal={this.closePopup}
            />
          </Modal>
        </GFooterDiv>
      </div>
    );
  }
}

QuoteStickyFooter.propTypes = {
  tripId: PropTypes.number.isRequired,
  quoteId: PropTypes.number.isRequired,
  quoteStatus: PropTypes.shape({
    isRejected: PropTypes.bool,
    isInvoiced: PropTypes.bool,
    isInvoiceRequested: PropTypes.bool,
    isCancelled: PropTypes.bool,
    isConverted: PropTypes.bool,
    isRequestCancelled: PropTypes.bool
  }).isRequired,
  currentQuote: PropTypes.object.isRequired,
  convertedQuote: PropTypes.object,
  quoteRejectionReasons: PropTypes.object.isRequired,
  userDetails: PropTypes.object.isRequired,

  bookNow: PropTypes.func.isRequired,
  rejectQuote: PropTypes.func.isRequired,
  reactivateQuote: PropTypes.func.isRequired,
  reactivateBooking: PropTypes.func.isRequired,
  cancelBooking: PropTypes.func.isRequired,

  rejectQuoteSuccess: PropTypes.bool.isRequired,
  reactivateQuoteSuccess: PropTypes.bool.isRequired,

  trackSegment: PropTypes.func,
  showRoster: PropTypes.func,
  setPdmCategory: PropTypes.func.isRequired,
  showHidePdmView: PropTypes.func.isRequired
};
QuoteStickyFooter.defaultProps = {
  convertedQuote: {},
  trackSegment: () => {},
  showRoster: () => {}
};

export default QuoteStickyFooter;
