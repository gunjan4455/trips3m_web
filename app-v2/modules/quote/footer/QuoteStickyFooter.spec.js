import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import QuoteStickyFooter from './QuoteStickyFooter';

import CancelBooking from 'modules/trip/ctas/CancelBooking';
import ReactivateTrip from 'modules/trip/ctas/ReactivateTrip';
import ReactivateQuote from '../ctas/ReactivateQuote';
import SocialShare from '../ctas/SocialShare';
import SwitchQuote from '../ctas/SwitchQuote';
import PayNow from '../ctas/PayNow';
import BookNow from '../ctas/BookNow';
import ViewInvoiceAndPay from '../ctas/ViewInvoiceAndPay';
import RejectQuote from '../ctas/RejectQuote';

import QUOTE_EVENTS_ENUMS from 'modules/quote/EventTypes';
import TRIP_EVENTS_ENUMS from 'modules/trip/ChildEventsType';

configure({ adapter: new Adapter() });

const BookNowMock = jest.fn();
const RejectQuoteMock = jest.fn();
const ReactivateQuoteMock = jest.fn();
const ReactivateBookingMock = jest.fn();
const CancelBookingMock = jest.fn();

const DEFAULT_PROPS = {
  tripId: 3918699,
  quoteId: 7124217,
  currentQuote: {
    installmentInfo: {
      isAllPaymentReceived: false,
      installmentsReceived: 0,
      nextScheduledDate: null,
    },
    invoiceShowUrl: '',
    opDetails: {
      id: 2251659,
      name: 'Damini Oberoi',
      phoneNo: '18001235555',
      email: 'damini.oberoi@traveltriangle.com',
      image: '/pics/original/missing.png',
      initials: 'D',
    },
  },
  convertedQuote: {
    id: 5,
  },
  quoteStatus: {
    isCancelled: false,
    isConverted: false,
    isInvoiceRequested: false,
    isInvoiced: false,
    isRejected: false,
    isRequestCancelled: false,
  },
  quoteRejectionReasons: {
    '1': 'Quotes are higher than my budget',
    '2': 'Want a different itinerary',
    '3': 'Wish to customize my existing package',
    '4': 'Want to travel to some other destination',
    '5': 'Others',
    '6': 'Not satisfied with the hotels',
  },
  userDetails: {
    first_name: 'Rahul',
    last_name: 'Jain',
  },
  bookNowSuccess: false,
  rejectQuoteSuccess: false,
  reactivateQuoteSuccess: false,
  bookNow: BookNowMock,
  rejectQuote: RejectQuoteMock,
  reactivateQuote: ReactivateQuoteMock,
  reactivateBooking: ReactivateBookingMock,
  cancelBooking: CancelBookingMock,
};

const shallowWrappperQuoteFooterComponent = (props) => {
  return shallow(<QuoteStickyFooter {...props} />);
};

describe('Sticky footer: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS, quoteStatus: { ...DEFAULT_PROPS.quoteStatus } };

  afterEach(() => {
    props = { ...DEFAULT_PROPS, quoteStatus: { ...DEFAULT_PROPS.quoteStatus } };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperQuoteFooterComponent(props);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Sticky footer: Should render correct cta\'s ', () => {
  let props = {
    ...DEFAULT_PROPS,
    quoteStatus: { ...DEFAULT_PROPS.quoteStatus },
    currentQuote: {
      ...DEFAULT_PROPS.currentQuote,
      installmentInfo: { ...DEFAULT_PROPS.currentQuote.installmentInfo }
    }
  };

  afterEach(() => {
    props = {
      ...DEFAULT_PROPS, quoteStatus: { ...DEFAULT_PROPS.quoteStatus },
      currentQuote: {
        ...DEFAULT_PROPS.currentQuote,
        installmentInfo: { ...DEFAULT_PROPS.currentQuote.installmentInfo }
      }
    };
  });

  it("Rejected Quote: should render Share, Reactivate Quote cta for 'Rejected Quote'", () => {
    props.quoteStatus.isRejected = true;
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(ReactivateQuote)).toHaveLength(1);
  });

  it("Canceled Booking: should render Share, Reactivate Quote cta for Canceled Booking", () => {
    props.quoteStatus.isRequestCancelled = true;
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(ReactivateTrip)).toHaveLength(1);
  });

  it("Quote cancelled: should render Share, Switch Quote for Quote cancelled, as another quote is confirmed", () => {
    props.quoteStatus.isCancelled = true;
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(SwitchQuote)).toHaveLength(1);
  });

  it("Quote Converted: should render Share, ViewInvoiceAndPay for all installments paid", () => {
    props.quoteStatus.isConverted = true;
    props.currentQuote.installmentInfo.isAllPaymentReceived = true;
    props.currentQuote.invoiceShowUrl = 'test url';

    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(ViewInvoiceAndPay)).toHaveLength(1);
  });

  it("Quote Converted: should render Share, Cancel booking and ViewInvoiceAndPay for atlease one installments paid", () => {
    props.quoteStatus.isConverted = true;
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(CancelBooking)).toHaveLength(1);
    expect(wrapper.find(ViewInvoiceAndPay)).toHaveLength(1);
  });

  it("Invoice Received: should render Share, Reject Quote and View Invoice and Pay cta", () => {
    props.quoteStatus.isInvoiced = true;
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(RejectQuote)).toHaveLength(1);
    expect(wrapper.find(ViewInvoiceAndPay)).toHaveLength(1);
  });

  it("Invoice Requested: should render Share, Reject Quote and Proceed to pay cta for 'Invoice Requested'", () => {
    props.quoteStatus.isInvoiceRequested = true;
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(RejectQuote)).toHaveLength(1);
    expect(wrapper.find(PayNow)).toHaveLength(1);
  });

  it("Book Now: should render Share, Book Now cta", () => {
    const wrapper = shallowWrappperQuoteFooterComponent(props);

    expect(wrapper.find(SocialShare)).toHaveLength(1);
    expect(wrapper.find(BookNow)).toHaveLength(1);
  });
});


describe('Sticky footer: callbackHandler ', () => {
  let props = { ...DEFAULT_PROPS, quoteStatus: { ...DEFAULT_PROPS.quoteStatus } };
  const wrapper = shallowWrappperQuoteFooterComponent(props);

  afterEach(() => {
    props = { ...DEFAULT_PROPS, quoteStatus: { ...DEFAULT_PROPS.quoteStatus } };
  });


  it("should call booknow props with correct data", () => {

    wrapper.instance().callbackHandler(QUOTE_EVENTS_ENUMS.BOOK_NOW, {});

    expect(BookNowMock).toHaveBeenCalledWith({
      tripId: 3918699,
      quoteId: 7124217
    });
  });

  it("should call rejectQuote with correct data'", () => {

    wrapper.instance().callbackHandler(QUOTE_EVENTS_ENUMS.REJECT_QUOTE, {});

    expect(BookNowMock).toHaveBeenCalledWith({
      tripId: 3918699,
      quoteId: 7124217
    });
  });

  it("should call reactivateQuote props with correct data'", () => {

    wrapper.instance().callbackHandler(QUOTE_EVENTS_ENUMS.REACTIVATE_QUOTE, {});

    expect(ReactivateQuoteMock).toHaveBeenCalledWith({
      tripId: 3918699,
      quoteId: 7124217
    });
  });

  it("should call reactivateBooking props with correct data'", () => {

    wrapper.instance().callbackHandler(TRIP_EVENTS_ENUMS.REACTIVATE_TRIP, {});

    expect(ReactivateBookingMock).toHaveBeenCalledWith({
      tripId: 3918699,
      quoteId: 7124217
    });
  });

  it("should call cancelBooking props with correct data'", () => {

    wrapper.instance().callbackHandler(TRIP_EVENTS_ENUMS.CANCEL_TRIP, {});

    expect(CancelBookingMock).toHaveBeenCalledWith({
      tripId: 3918699,
      quoteId: 7124217
    });
  });

});
