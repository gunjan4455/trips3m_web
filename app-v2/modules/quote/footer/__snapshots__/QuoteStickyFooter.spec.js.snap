// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticky footer: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuoteStickyFooter
    bookNow={[MockFunction]}
    bookNowSuccess={false}
    cancelBooking={[MockFunction]}
    convertedQuote={
      Object {
        "id": 5,
      }
    }
    currentQuote={
      Object {
        "installmentInfo": Object {
          "installmentsReceived": 0,
          "isAllPaymentReceived": false,
          "nextScheduledDate": null,
        },
        "invoiceShowUrl": "",
        "opDetails": Object {
          "email": "damini.oberoi@traveltriangle.com",
          "id": 2251659,
          "image": "/pics/original/missing.png",
          "initials": "D",
          "name": "Damini Oberoi",
          "phoneNo": "18001235555",
        },
      }
    }
    quoteId={7124217}
    quoteRejectionReasons={
      Object {
        "1": "Quotes are higher than my budget",
        "2": "Want a different itinerary",
        "3": "Wish to customize my existing package",
        "4": "Want to travel to some other destination",
        "5": "Others",
        "6": "Not satisfied with the hotels",
      }
    }
    quoteStatus={
      Object {
        "isCancelled": false,
        "isConverted": false,
        "isInvoiceRequested": false,
        "isInvoiced": false,
        "isRejected": false,
        "isRequestCancelled": false,
      }
    }
    reactivateBooking={[MockFunction]}
    reactivateQuote={[MockFunction]}
    reactivateQuoteSuccess={false}
    rejectQuote={[MockFunction]}
    rejectQuoteSuccess={false}
    showRoster={[Function]}
    trackSegment={[Function]}
    tripId={3918699}
    userDetails={
      Object {
        "first_name": "Rahul",
        "last_name": "Jain",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ForwardRef(render)
        className="flex sbcw p8 bs6 at_quoteFooterSection"
      >
        <ExplainTheQuote
          handleChatClick={[Function]}
          isQuoteCardCta={false}
        />
        <ModalComponent
          customClass=""
          customCloseTrigger={true}
          disabled={false}
          fullView={true}
          hasCustomClose={true}
          hasNoClose={false}
          isOpen={false}
          isTriggerFullView={false}
          isWhiteIcon={false}
          onAfterModalOpen={[Function]}
          onRequestCloseCustom={[Function]}
          trigger={null}
        >
          <BookNowPopup
            closeModal={[Function]}
            opsDetail={
              Object {
                "email": "damini.oberoi@traveltriangle.com",
                "id": 2251659,
                "image": "/pics/original/missing.png",
                "initials": "D",
                "name": "Damini Oberoi",
                "phoneNo": "18001235555",
              }
            }
            trackSegment={[Function]}
          />
        </ModalComponent>
      </ForwardRef(render)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          Array [
            <ExplainTheQuote
              handleChatClick={[Function]}
              isQuoteCardCta={false}
            />,
          ],
          <ModalComponent
            customClass=""
            customCloseTrigger={true}
            disabled={false}
            fullView={true}
            hasCustomClose={true}
            hasNoClose={false}
            isOpen={false}
            isTriggerFullView={false}
            isWhiteIcon={false}
            onAfterModalOpen={[Function]}
            onRequestCloseCustom={[Function]}
            trigger={null}
          >
            <BookNowPopup
              closeModal={[Function]}
              opsDetail={
                Object {
                  "email": "damini.oberoi@traveltriangle.com",
                  "id": 2251659,
                  "image": "/pics/original/missing.png",
                  "initials": "D",
                  "name": "Damini Oberoi",
                  "phoneNo": "18001235555",
                }
              }
              trackSegment={[Function]}
            />
          </ModalComponent>,
        ],
        "className": "flex sbcw p8 bs6 at_quoteFooterSection",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "explain-the-quote",
          "nodeType": "class",
          "props": Object {
            "handleChatClick": [Function],
            "isQuoteCardCta": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <BookNowPopup
              closeModal={[Function]}
              opsDetail={
                Object {
                  "email": "damini.oberoi@traveltriangle.com",
                  "id": 2251659,
                  "image": "/pics/original/missing.png",
                  "initials": "D",
                  "name": "Damini Oberoi",
                  "phoneNo": "18001235555",
                }
              }
              trackSegment={[Function]}
            />,
            "customClass": "",
            "customCloseTrigger": true,
            "disabled": false,
            "fullView": true,
            "hasCustomClose": true,
            "hasNoClose": false,
            "isOpen": false,
            "isTriggerFullView": false,
            "isWhiteIcon": false,
            "onAfterModalOpen": [Function],
            "onRequestCloseCustom": [Function],
            "trigger": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "closeModal": [Function],
              "opsDetail": Object {
                "email": "damini.oberoi@traveltriangle.com",
                "id": 2251659,
                "image": "/pics/original/missing.png",
                "initials": "D",
                "name": "Damini Oberoi",
                "phoneNo": "18001235555",
              },
              "trackSegment": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          Object {
            "bottom": "0",
            "justifyContent": "space-between",
            "left": "0",
            "minHeight": "48px",
            "position": "fixed",
            "right": "0",
            "zIndex": "9",
          },
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef(render)
          className="flex sbcw p8 bs6 at_quoteFooterSection"
        >
          <ExplainTheQuote
            handleChatClick={[Function]}
            isQuoteCardCta={false}
          />
          <ModalComponent
            customClass=""
            customCloseTrigger={true}
            disabled={false}
            fullView={true}
            hasCustomClose={true}
            hasNoClose={false}
            isOpen={false}
            isTriggerFullView={false}
            isWhiteIcon={false}
            onAfterModalOpen={[Function]}
            onRequestCloseCustom={[Function]}
            trigger={null}
          >
            <BookNowPopup
              closeModal={[Function]}
              opsDetail={
                Object {
                  "email": "damini.oberoi@traveltriangle.com",
                  "id": 2251659,
                  "image": "/pics/original/missing.png",
                  "initials": "D",
                  "name": "Damini Oberoi",
                  "phoneNo": "18001235555",
                }
              }
              trackSegment={[Function]}
            />
          </ModalComponent>
        </ForwardRef(render)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            Array [
              <ExplainTheQuote
                handleChatClick={[Function]}
                isQuoteCardCta={false}
              />,
            ],
            <ModalComponent
              customClass=""
              customCloseTrigger={true}
              disabled={false}
              fullView={true}
              hasCustomClose={true}
              hasNoClose={false}
              isOpen={false}
              isTriggerFullView={false}
              isWhiteIcon={false}
              onAfterModalOpen={[Function]}
              onRequestCloseCustom={[Function]}
              trigger={null}
            >
              <BookNowPopup
                closeModal={[Function]}
                opsDetail={
                  Object {
                    "email": "damini.oberoi@traveltriangle.com",
                    "id": 2251659,
                    "image": "/pics/original/missing.png",
                    "initials": "D",
                    "name": "Damini Oberoi",
                    "phoneNo": "18001235555",
                  }
                }
                trackSegment={[Function]}
              />
            </ModalComponent>,
          ],
          "className": "flex sbcw p8 bs6 at_quoteFooterSection",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "explain-the-quote",
            "nodeType": "class",
            "props": Object {
              "handleChatClick": [Function],
              "isQuoteCardCta": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <BookNowPopup
                closeModal={[Function]}
                opsDetail={
                  Object {
                    "email": "damini.oberoi@traveltriangle.com",
                    "id": 2251659,
                    "image": "/pics/original/missing.png",
                    "initials": "D",
                    "name": "Damini Oberoi",
                    "phoneNo": "18001235555",
                  }
                }
                trackSegment={[Function]}
              />,
              "customClass": "",
              "customCloseTrigger": true,
              "disabled": false,
              "fullView": true,
              "hasCustomClose": true,
              "hasNoClose": false,
              "isOpen": false,
              "isTriggerFullView": false,
              "isWhiteIcon": false,
              "onAfterModalOpen": [Function],
              "onRequestCloseCustom": [Function],
              "trigger": null,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "closeModal": [Function],
                "opsDetail": Object {
                  "email": "damini.oberoi@traveltriangle.com",
                  "id": 2251659,
                  "image": "/pics/original/missing.png",
                  "initials": "D",
                  "name": "Damini Oberoi",
                  "phoneNo": "18001235555",
                },
                "trackSegment": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "bottom": "0",
              "justifyContent": "space-between",
              "left": "0",
              "minHeight": "48px",
              "position": "fixed",
              "right": "0",
              "zIndex": "9",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
